test/DeathTest.cpp

```diff
@@ -144,7 +147,7 @@ TEST(DeathTest, ThreadSafeTest) {
       Death::Instance().RegisterDeathEvent(&DeathTest::RaceTest, "race");
    };
    std::vector<std::future<void>> waitingPromises;
-   for (int i = 0; i < 10000; i++) {
+   for (int i = 0; i < 10; i++) {
```

Comment:
Are we actually testing if it'll work at all? Consider this happens on the build system and not the destination system?